### PR TITLE
Proof of concept "required" magic comment

### PR DESCRIPTION
**Pull request by jocmp**
Adds a `gql-required` magic comment to allow protobuf fields to be nullable. This is in contrast to two built-in options:

1. Custom descriptors. This would require syncing an override value across repos to allow the protoc generator to register the extension

2. "optional" keyword. This is available on message types, meaning that repeated fields cannot be made nullable.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
+all: build
+
 clean:
 	@rm -rf target/
 
 build: clean
-	@env GOOS=linux GOARCH=amd64 go build $(GOFLAGS) -o target/protoc-gen-graphql.linux
-	@env GOOS=darwin go build $(GOFLAGS) -o target/protoc-gen-graphql.darwin
+	@env GOOS=linux GOARCH=amd64 go build $(GOFLAGS) -o target/protoc-gen-graphql.linux.amd64
+	@env GOOS=linux GOARCH=arm64 go build $(GOFLAGS) -o target/protoc-gen-graphql.linux.arm64
```
Comment (benoist-folio) [inline Makefile:14]: Ya'll using arm/graviton these days?

Comment (jocmp) [reply Makefile:14]: yep! pairs nicely with those ARM macs

**Comment by benoist-folio** [comment]
Have you considered using the built in descriptors for this? They should be present on every platform:

https://protobuf.dev/reference/protobuf/google.protobuf/#string-value

They don't have the greatest DX, but they do just work.

**Comment by jocmp** [comment]
@benoist-folio i hadn't, but i'll definitely check it out. i closed this due to other factors that cropped up that will prevent nullability in the graph. i might be able to get back to it at some point.

**Comment by benoist-folio** [comment]
Makes sense. I often ran into similar problems with type mismatch between how developers think, what protobuf thinks, and what graphql thinks. `StringValue` and its friends make it way more obvious and you _could_ create a custom resolver for it so that it allows for explicitly `nullable` fields.

**Pull request closed without merging**